patroni/postgresql/misc.py

```diff
@@ -58,8 +58,12 @@ def parse_lsn(lsn):
     return int(t[0], 16) * 0x100000000 + int(t[1], 16)
 
 
-def parse_history(data):
-    for line in data.split('\n'):
+def parse_history(data, max_timelines=0):
+    timelines = None
+    if max_timelines is not None and abs(max_timelines) > 0:
+        # twice the requested as the timeline history has empty lines
+        timelines = -abs(max_timelines*2)
+    for line in data.split('\n')[timelines:]:
```

Comment:
1. I wouldn't rely on empty lines, because there is no guaranty that it will not change in the future.
2. I also think that passing the `max_timelines` argument all the way down to this function is unnecessary. Do we really want to save a few cpu cycles on parsing the complete history file? It would be good enough to just change the https://github.com/zalando/patroni/pull/1491/files#diff-57c53e988b775ee2902488fb33c1f433R525 to
```python
               for line in history[-max_timelines:]:
```

Reply:
1. Agree. We can move that logic to get_history to trim timelines.
2. Will update.